pkg/ingester/ingester_claim.go

```diff
@@ -38,6 +38,22 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 		return err
 	}
 
+	// The ingesters state effectively works as a giant mutex around this whole
+	// method, and as such we have to ensure we unlock the mutex.
+	defer func() {
+		errc := make(chan error)
+		i.actorChan <- func() {
+			if i.state == ring.JOINING {
+				errc <- i.changeState(ring.PENDING)
```

Comment:
This is guaranteed to fail, because `changeState` doesn't permit the state to be changed to `PENDING`:

```go
	if !((i.state == ring.PENDING && state == ring.JOINING) || // triggered by ClaimStart
		(i.state == ring.PENDING && state == ring.ACTIVE) || // triggered by autoJoin
		(i.state == ring.JOINING && state == ring.ACTIVE) || // triggered by ClaimFinish
		(i.state == ring.ACTIVE && state == ring.LEAVING)) { // triggered by shutdown
		return fmt.Errorf("Changing ingester state from %v -> %v is disallowed", i.state, state)
	}
```